app/controllers/concerns/foreman/controller/discovered_extensions.rb

```diff
@@ -78,11 +79,17 @@ def perform_reboot_all hosts = Host::Discovered.all
             error =  "#{discovered_host.name}: failed to reboot "
             overall_errors << error
             logger.error error
+            host_details << {:name => discovered_host.name, :error => error}
           end
         rescue Exception => e
-          error = "#{discovered_host.name}: #{e.to_s} "
-          overall_errors << error
-          logger.error error
+          msg = if e.respond_to?(:wrapped_exception)
+                  e.wrapped_exception.message
+                else
+                  e.message
+                end
+          overall_errors << msg
+          logger.error msg
```

Comment:
Please use: `Foreman::Logging.exception` helper, see in core. This only puts the message in logs, we want message and backtrace.